store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -663,9 +663,27 @@ public long getMaxOffsetInQueue(String topic, int queueId) {
     }
 
     public long getMinOffsetInQueue(String topic, int queueId) {
+        return getMinOffsetInQueue(topic, queueId,false);
+    }
+
+    public long getMinOffsetInQueue(String topic, int queueId, boolean realOffset) {
         ConsumeQueue logic = this.findConsumeQueue(topic, queueId);
         if (logic != null) {
-            return logic.getMinOffsetInQueue();
+            long consumeQueueOffset = logic.getMinOffsetInQueue();
+            MessageExt msgExt = null;
+            long commitLogOffset = 0L;
+            if (realOffset) {
+                for (;consumeQueueOffset <= getMaxOffsetInQueue(topic,queueId);consumeQueueOffset++) {
+                    commitLogOffset = getCommitLogOffsetInQueue(topic, queueId, consumeQueueOffset);
+                    msgExt = lookMessageByOffset(commitLogOffset);
+                    if (null != msgExt) {
+                        return consumeQueueOffset;
+                    }
+                }
+                return consumeQueueOffset;
```

Comment:
IMO, this is a very CPU intensive operation.  It is possible to get a lot of cold data and decode it, which will affect the stability of the broker.

Reply:
sorry，because before we need to query the content of the dead letter queue message，did‘t consider this place does not need。I am modified to compare 'this.commitLog.getMinOffset()'